libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -704,4 +736,74 @@ std::shared_ptr<arrow::DataType> make_experimental_type(const type& t) {
   return caf::visit(f, t);
 }
 
+type make_vast_type(const std::shared_ptr<arrow::DataType>& arrow_type) {
+  switch (arrow_type->id()) {
+    case arrow::Type::NA:
+      return type{none_type{}};
+    case arrow::Type::BOOL:
+      return type{bool_type{}};
+    case arrow::Type::INT64:
+      return type{integer_type{}};
+    case arrow::Type::UINT64:
+      return type{count_type{}};
+    case arrow::Type::DOUBLE:
+      return type{real_type{}};
+    case arrow::Type::DURATION:
+      return type{duration_type{}};
+    case arrow::Type::STRING:
+      return type{string_type{}};
+    case arrow::Type::TIMESTAMP:
+      return type{time_type{}};
+    case arrow::Type::FIXED_SIZE_BINARY: {
+      const auto& t
+        = std::static_pointer_cast<arrow::FixedSizeBinaryType>(arrow_type);
+      switch (auto width = t->byte_width(); width) {
+        case 16:
+          return type{address_type{}};
+        case 17:
+          return type{subnet_type{}};
+        default:
+          VAST_ASSERT(
+            false,
+            fmt::format("unhandled Arrow type: FIXEDBINARY[{}]", width).c_str());
+      }
+      return type{time_type{}};
+    }
+    case arrow::Type::LIST: {
+      const auto& t = std::static_pointer_cast<arrow::ListType>(arrow_type);
+      const auto& embedded_type = make_vast_type(t->value_type());
+      return type{list_type{embedded_type}};
+    }
+    case arrow::Type::STRUCT: {
+      std::vector<struct record_type::field> field_types;
+      field_types.reserve(arrow_type->num_fields());
+      for (const auto& f : arrow_type->fields()) {
+        field_types.emplace_back(f->name(), make_vast_type(*f));
+      }
+
+      return type{record_type{field_types}};
+    }
+
+    default:
+      VAST_ASSERT(
+        false,
+        fmt::format("unhandled Arrow type: {}", arrow_type->ToString()).c_str());
+      return type{none_type{}};
+  }
+}
+
+type make_vast_type(const arrow::Field& arrow_field) {
+  return make_vast_type(arrow_field.type());
+}
```

Comment:
```suggestion
struct record_type::field make_vast_type(const arrow::Field& arrow_field) {
  return {arrow_field.name(), make_vast_type(arrow_field.type())};
}
```

Otherwise we lose information from the field that we may need to later construct a `record_type` from it.

Note the difference between `record_type::field` and `record_type::field_view`: The former stores the name as a string, the latter only contains a string_view whose ownership is bound to the type. That's why you usually want to pass the latter, and return the former if you create a new one.

Reply:
done, thanks